### ISSUE TITLE
Unsupported mail transport [sendinblue | mailjet | ... ].

### ISSUE THREAD

**Issue by RoodFruit** [laravel/framework#40330]
- Laravel Version: 9.x
- PHP Version: 8.1.1
- Database Driver & Version: -

### Description:
Sendinblue is implemented by Laravel and the mailer can be required by `symfony/sendinblue-mailer`, but Laravel does not detect the transport yet. After some digging, I found out that we're missing the function `createSendinblueTransport()` in the `Illuminate\Mail\MailManager`.

Other mailers have the necessary transport and thus the function to create the instance of the transport driver. For example `createMailgunTransport()`, `createPostmarkTransport()` and `createSesTransport()`.

### Steps To Reproduce:

1. `composer require symfony/sendinblue-mailer`
2. Edit your `config/mail.php` by adding:
```
        'sendinblue' => [
            'transport' => 'sendinblue',
        ],
```
3. Edit your `config/services.php` by adding:
```
    'sendinblue' => [
        // this does not matter yet for this issue, since 
        // the creation of the transport is not there yet!
        'api' => 'xxx',
    ],
```

### TLDR;

I know Sendinblue is not a default configured mail driver in Laravel, but with the introduction of the Symfony Mailer, it should be as easy as adding the correct transport in `config/mail.php` and the necessary variables in `config/services.php` for the whole list of existing symfony mailers.

The current list of symfony mailers is:
```
Amazon SES | composer require symfony/amazon-mailer
Gmail      | composer require symfony/google-mailer
MailChimp  | composer require symfony/mailchimp-mailer
Mailgun    | composer require symfony/mailgun-mailer
Mailjet    | composer require symfony/mailjet-mailer
Postmark   | composer require symfony/postmark-mailer
SendGrid   | composer require symfony/sendgrid-mailer
Sendinblue | composer require symfony/sendinblue-mailer
OhMySMTP   | composer require symfony/oh-my-smtp-mailer
```

**Comment by driesvints** [comment]
You can add this transport as a custom transport. I see we're not documenting that yet so I'll try to write some docs on that soon. Welcoming PR's however, thanks.

**Comment by RoodFruit** [comment]
> You can add this transport as a custom transport. I see we're not documenting that yet so I'll try to write some docs on that soon. Welcoming PR's however, thanks.

Ah.. I didn't know. Can you already share a quick snippet of how-to, so me and my team can continue on the upgrades of our systems to Laravel 9?

**Comment by RoodFruit** [comment]
@driesvints you got a heads-up or a snippet on how-to?

**Comment by driesvints** [comment]
@RoodFruit bit swamped now with the upcoming Laravel 9 release. I'll try to get to that soon but in the meantime you might just figure it out for yourself. It's basically adding a `Mail::extend()` call in a service provider to register a new transport that you can use. Try looking inside the MailManager class how the other ones are registered.

**Comment by RoodFruit** [comment]
@driesvints that's what I was looking for. Thanks for the kickstart!

**Comment by RoodFruit** [comment]
For those that have the same problem, the following works on our end.

```
<?php

namespace App\Providers;

use Illuminate\Mail\MailManager;
use Illuminate\Support\ServiceProvider;
use Symfony\Component\Mailer\Bridge\Sendinblue\Transport\SendinblueTransportFactory;
use Symfony\Component\Mailer\Transport\Dsn;

class AppServiceProvider extends ServiceProvider
{
    /**
     * Register any application services.
     */
    public function register(): void
    {
        //
    }

    /**
     * Bootstrap any application services.
     */
    public function boot(): void
    {
        $this->app[MailManager::class]->extend('sendinblue', function () {
            return (new SendinblueTransportFactory())->create(
                new Dsn('sendinblue+api', 'default', $this->app['config']->get('services.sendinblue.api'))
            );
        });
    }
}
```

Where `$this->app['config']->get('services.sendinblue.api')` implies that in the config `services.php` we have defined the following:

```
<?php

return [

    ...

    'sendinblue' => [
        'api' => env('SENDINBLUE_API_KEY'),
    ],

];

```

@driesvints If we're doing something wrong or if you have any pointers, I'm open to suggestions :).

**Comment by driesvints** [comment]
Lgtm 👍

**Comment by driesvints** [comment]
@RoodFruit we got this one in: https://github.com/laravel/docs/pull/7644

**Comment by khaledelmahdi** [comment]
Hi,

I am getting this error when I try to register SendGridTransport:

***Mailer [sendgrid] is not defined.***

I've added the package:

```
composer require sendgrid/sendgrid
```

I've edited the `AppServiceProvider` file:

```php
<?php

namespace App\Providers;

use Illuminate\Support\Facades\Mail;
use Illuminate\Support\ServiceProvider;
use Symfony\Component\Mailer\Bridge\Sendgrid\Transport\SendgridTransportFactory;
use Symfony\Component\Mailer\Transport\Dsn;

class AppServiceProvider extends ServiceProvider
{
    public function register()
    {
        //
    }

    public function boot()
    {
        Mail::extend('sendgrid', function () {
            return (new SendgridTransportFactory)->create(
                new Dsn(
                    'sendgrid+api',
                    'default',
                    config('services.sendgrid.key')
                )
            );
        });
    }
}

```

I've edited the `config\services.php` file:

```
'sendgrid' => [
        'key' => env('SENDGRID_API_KEY'),
],
```

What I am doing wrong here?
